### PR TITLE
Add create new todo using <form>

### LINKED ISSUE #79: Adding a new to-do with the help of <form>

### PR DESCRIPTION
**Pull request by mjzac**
Closes #79

### CODE REVIEW

File: html-forms/add-new-todo/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+## Text
+
+In this lesson, we will use our newly designed UI to create new to-do items. To do that, we first need to make our backend handle such a request.
```
Comment (avishekjana) [inline html-forms/add-new-todo/README.md:3]: In this lesson, we will use our newly designed UI to create new to-do items. To do that, first we have to update our backend to handle such requests.

File: html-forms/add-new-todo/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+```js
+router.get("/", async function (req, res, next) {
+  const overdue = await db.Todo.overdue();
+  const dueToday = await db.Todo.dueToday();
+  const dueLater = await db.Todo.dueLater();
+  res.render("index", { overdue, dueToday, dueLater });
+});
+```
+
+To create a new todo, we browser should send a `POST` request. The required parameters like `title` and `dueDate` are sent as body of the `POST` request. To create such a handler we need to declare it using `router.post`. Express.js would parse the body of the request and make them available at `req.body`. And we would be able to extract `title` and `dueDate` from `req.body`.
```
Comment (avishekjana) [inline html-forms/add-new-todo/README.md:24]: To create a new todo, web browser should send a `POST` request. The required parameters like `title` and `dueDate` are sent as body of the `POST` request. To create such a handler, we need to declare it using `router.post`. Express.js would parse the body of the request and make them available at `req.body`. And we would be able to extract `title` and `dueDate` from `req.body`.

File: html-forms/add-new-todo/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+Let's edit the `todos.ejs` file to make sure the new todo item form has an `action` and `method` attribute.
+
+```html
+<form action="/todos" method="POST">...</form>
+```
+
+This makes sure that when the `Add` button is clicked, the form will be submitted as a `POST` request to `/todos` endpoint.
+
+Next, we need to make sure the todo text is sent to server using the key `title` and the due date using the key `dueDate`. We can do this by adding `name` attribute on respective input fields. So the final for looks like:
```
Comment (avishekjana) [inline html-forms/add-new-todo/README.md:46]: Next, we need to make sure the todo text is sent to the server using the key `title` and the due date using the key `dueDate`. We can do this by adding `name` attribute on respective input fields. So, the final form looks like:

File: html-forms/add-new-todo/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        class="text-white bg-green-600 hover:bg-green-700 font-medium rounded text-sm px-5 py-2 mr-2 mb-2"
+      >
+        Add
+      </button>
+    </div>
+  </div>
+</form>
+```
+
+Save the file and restart the server. We should now be able to create new todo items from the user interface.
```
Comment (avishekjana) [inline html-forms/add-new-todo/README.md:80]: Save the file and restart the server. Now we would be able to create new todo items from the user interface.

File: html-forms/add-new-todo/README.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
+## Text
+
+In this lesson, we will use our newly designed UI to create new to-do items. To do that, we first need to make our backend handle such a request.
+
+Edit the `todos.js` in `router` folder with following contents.
```
Comment (avishekjana) [inline html-forms/add-new-todo/README.md:5]: @mjzac All of a sudden, the `router` folder appears here. You need to guide the learner to create the folder structure.
And the routes are already been defined in level 6 lesson 3. Please check if you can build on top of that.
https://github.com/pupilfirst/wd201-scripts/pull/18/files

Comment (mjzac) [reply html-forms/add-new-todo/README.md]: I have removed the router and used the same routes from previous level.

**Pull request merged**